support-frontend/assets/helpers/async/fetch.ts

```diff
@@ -4,23 +4,24 @@ import 'helpers/csrf/csrfReducer';
 type Credentials = 'omit' | 'same-origin' | 'include';
 
 /** Sends a request to an API and converts the response into a JSON object */
-function fetchJson(
-	endpoint: string,
-	settings: Record<string, any>,
-): Promise<Record<string, any>> {
-	return fetch(endpoint, settings).then((resp) => resp.json());
+async function fetchJson(
+	endpoint: RequestInfo,
+	settings: RequestInit,
+): Promise<Record<string, unknown>> {
+	const resp = await fetch(endpoint, settings);
+	return (await resp.json()) as Record<string, unknown>;
 }
 
 /** Builds a `RequestInit` object for use with GET requests using the Fetch API */
 function getRequestOptions(
 	credentials: Credentials,
 	csrf: CsrfState | null,
-): Record<string, any> {
+): Record<string, unknown> {
```

Comment:
Given that we're in control of what we're returning, it might make sense to be more specific with this type? e.g.

```ts
interface GetRequestOptions {
  method: 'GET';
  headers: /* ... */;
  credentials: /* ... */;
}
```